spec.html

```diff
@@ -6281,38 +6281,23 @@ <h1>CreateIterResultObject ( _value_, _done_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-createlistiteratorRecord" oldids="sec-createlistiterator" aoid="CreateListIteratorRecord">
+    <emu-clause id="sec-createlistiteratorRecord" oldids="sec-createlistiterator,sec-listiteratornext-functions,sec-listiterator-next" aoid="CreateListIteratorRecord">
       <h1>CreateListIteratorRecord ( _list_ )</h1>
       <p>The abstract operation CreateListIteratorRecord takes argument _list_. It creates an Iterator (<emu-xref href="#sec-iterator-interface"></emu-xref>) object record whose next method returns the successive elements of _list_. It performs the following steps when called:</p>
       <emu-alg>
-        1. Let _iterator_ be OrdinaryObjectCreate(%IteratorPrototype%, &laquo; [[IteratedList]], [[ListNextIndex]] &raquo;).
-        1. Set _iterator_.[[IteratedList]] to _list_.
-        1. Set _iterator_.[[ListNextIndex]] to 0.
-        1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-listiteratornext-functions" title></emu-xref>.
-        1. Let _next_ be ! CreateBuiltinFunction(_steps_, &laquo; &raquo;).
-        1. Return Record { [[Iterator]]: _iterator_, [[NextMethod]]: _next_, [[Done]]: *false* }.
+        1. Let _closure_ be a new Abstract Closure with no parameters that captures _list_ and performs the following steps when called:
+          1. Let _index_ be 0.
+          1. Let _len_ be the number of elements of _list_.
+          1. Repeat, until _index_ &ge; _len_,
+            1. Set _index_ to _index_ + 1.
```

Comment:
Won't this yield the element at index 1 first? I think you need to swap this step and the next.

Reply:
Nice catch, thanks!